common.go

```diff
@@ -7,9 +7,11 @@ import (
 	"github.com/containers/image/types"
 )
 
-func getCopyOptions(reportWriter io.Writer) *cp.Options {
+func getCopyOptions(reportWriter io.Writer, credsContext *types.SystemContext) *cp.Options {
 	return &cp.Options{
-		ReportWriter: reportWriter,
+		ReportWriter:   reportWriter,
+		SourceCtx:      credsContext,
+		DestinationCtx: credsContext,
```

Comment:
I don't think we'll have a buildah command that will pass both a destination and a source systemcontext.  I think the command will not have any or will have either the source or the destination.  I'll send two params down to this function then, one for each type and will set them as needed from the calling function.